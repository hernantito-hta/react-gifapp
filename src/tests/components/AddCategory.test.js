import { shallow } from 'enzyme';
import AddCategory from '../../components/AddCategory';

describe('Testing component AddCategory', () => {
    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories = {setCategories} />);
    
    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories = {setCategories} />);
    });

    test('should render correctly the component', () => {
        
        expect(wrapper).toMatchSnapshot();
        
    })

    test('should change the textbox field function handleInputValue', () => {
        const input = wrapper.find('input');
        const value = 'Adding text 1';
        
        input.simulate('change', {
            target: { value }
        });

        expect(wrapper.find('p').text().trim()).toBe(value);
    })

    test('should not be called', () => {
        const form  = wrapper.find('form');
        form.simulate('submit', {
            preventDefault(){}
        });

        expect(setCategories).not.toHaveBeenCalled();
    })
    
    test('should call setCategories and clean text', () => {
        const value = 'Text description'
        let input = wrapper.find('input');
        input.simulate('change', {
            target: { value }
        });

        let form = wrapper.find('form');
        form.simulate('submit', {
            preventDefault(){}
        });
        
        expect(input.props().value).toBe('');
    })
})
