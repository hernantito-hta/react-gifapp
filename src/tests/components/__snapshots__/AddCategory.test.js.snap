// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component AddCategory should render correctly the component 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
